MdeModulePkg/Universal/BdsDxe/BdsEntry.c

```diff
@@ -417,16 +417,31 @@ BootBootOptions (
 
     PlatformBootManagerProcessBootCompletion (&BootOptions[Index]);        // MSCHANGE 00076 - record boot status
 
-    //
-    // If the boot via Boot#### returns with a status of EFI_SUCCESS, platform firmware
-    // supports boot manager menu, and if firmware is configured to boot in an
-    // interactive mode, the boot manager will stop processing the BootOrder variable and
-    // present a boot manager menu to the user.
-    //
-    if ((BootManagerMenu != NULL) && (BootOptions[Index].Status == EFI_SUCCESS)) {
-      EfiBootManagerBoot (BootManagerMenu);
-      break;
+    // MU_CHANGE [BEGIN] - Support infinite boot retries
+    if (!PcdGetBool (PcdSupportInfiniteBootRetries)) {
+      // MU_CHANGE [END] - Support infinite boot retries
+
+      //
+      // If the boot via Boot#### returns with a status of EFI_SUCCESS, platform firmware
+      // supports boot manager menu, and if firmware is configured to boot in an
+      // interactive mode, the boot manager will stop processing the BootOrder variable and
+      // present a boot manager menu to the user.
+      //
+      if ((BootManagerMenu != NULL) && (BootOptions[Index].Status == EFI_SUCCESS)) {
+        EfiBootManagerBoot (BootManagerMenu);
+        break;
+      }
+
+      // MU_CHANGE [BEGIN]- Support infinite boot retries
+    }
+
+    if (PcdGetBool (PcdSupportInfiniteBootRetries)) {
+      if (Index == (BootOptionCount -1)) {
+        Index = 0;

```

Comment:
This is not equivalent to assigning `Index` to `0` in the loop initialization step. On the next iteration, `Index` will be incremented and then compared against `BootOptionCount` with the value of `1`.

Is that intentional?

Reply:
It was an oversight. Updated to set Index to (UINTN)-1.